pkgs/development/tools/analysis/clazy/default.nix

```diff
@@ -0,0 +1,54 @@
+{
+    lib
+  , clangStdenv
+  , fetchFromGitHub
+  , llvmPackages
+  , cmake
+  , makeWrapper
+}:
+
+clangStdenv.mkDerivation rec {
+  pname = "clazy";
+  version = "1.11";
+
+  src = fetchFromGitHub {
+    owner  = "KDE";
+    repo   = "clazy";
+    rev    = "v${version}";
+    sha256 = "sha256-kcl4dUg84fNdizKUS4kpvIKFfajtTRdz+MYUbKcMFvg=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    makeWrapper
+    llvmPackages.clang
+    llvmPackages.llvm
+    llvmPackages.libclang
```

Comment:
I think llvmPackages.llvm/libclang should be  buildInputs

Reply:
I think you are right after re-reading that section of the nixpkgs manual. The different derivation input types are still a bit opaque to me.

Reply:
https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/stdenv.chapter.md#depsbuildbuild-var-stdenv-depsbuildbuild
If the dependency doesn’t care about the target platform (i.e. isn’t a compiler or similar tool), put it in nativeBuildInputs instead. So clang as a compiler should be `depsBuildBuild`,

but If we use `clangStdenv`,  it should not necessary to add llvmPackages.clang to nativeBuildInputs/depsBuildBuild，as we never add `gcc` to nativeBuildInputs when use `stdenv`

Reply:
I think I got it now. I was worried that since `clang++` was a runtime dependency that it would not be pulled in without explicitly defining it as a input.

Reply:
[Nix automatically computes all the runtime dependencies of a derivation](https://nixos.org/guides/nix-pills/automatic-runtime-dependencies.html), although I'm not aware of the implementation details, I believe this is the correct way

```
nix-store -q --references /nix/store/p4gwp54jjrbzi24ddgm1qz7g354kj7w2-clazy-1.11                                                                                                            main
...
/nix/store/w2nhdkmich4l416ip426qmcw9jja5f8z-clang-wrapper-14.0.6
/nix/store/p4gwp54jjrbzi24ddgm1qz7g354kj7w2-clazy-1.11
```